src/GafferSceneUIModule/SetEditorBinding.cpp

```diff
@@ -407,6 +407,422 @@ class SetNameColumn : public StandardPathColumn
+					{
+						excludedSetMemberCount++;
+					}
+				}
+			}
+
+			iconData->writable()["state:highlighted"] = g_setIncludedHighlightedIconName;
+			const bool allSetMembersIncluded = includedSetMembers.size() == setMembers->readable().size();
+			if( excludedSetMemberCount == 0 )
+			{
+				iconData->writable()["state:normal"] = allSetMembersIncluded ? g_setIncludedIconName : g_setPartiallyIncludedIconName;
+				result.toolTip = allSetMembersIncluded ? g_setIncludedToolTip : g_setPartiallyIncludedToolTip;
+			}
+			else if( includedSetMembers.size() == excludedSetMemberCount )
+			{
+				iconData->writable()["state:normal"] = g_setIncludedDisabledIconName;
+				result.toolTip = allSetMembersIncluded ? g_setIncludedOverrideToolTip : g_setPartiallyIncludedOverrideToolTip;
+			}
+			else
+			{
+				iconData->writable()["state:normal"] = g_setPartiallyDisabledIconName;
+				result.toolTip = allSetMembersIncluded ? g_setIncludedPartialOverrideToolTip : g_setPartiallyIncludedPartialOverrideToolTip;
+			}
+
+			return result;
+		}
+
+		CellData headerData( const IECore::Canceller *canceller ) const override
+		{
+			const auto visibleSet = ContextAlgo::getVisibleSet( m_context.get() );
+			return CellData( /* value = */ nullptr, /* icon = */ visibleSet.inclusions.isEmpty() ? g_inclusionsEmptyIconName : g_setIncludedIconName, /* background = */ nullptr, /* tooltip = */ new StringData( "Visible Set Inclusions" ) );
+		}
+
+	private :
+
+		void contextChanged( const IECore::InternedString &name )
+		{
+			if( ContextAlgo::affectsVisibleSet( name ) )
+			{
+				changedSignal()( this );
+			}
+		}
+
+		bool buttonPress( const ButtonEvent &event )
+		{
+			if( event.buttons != ButtonEvent::Left )
+			{
+				return false;
+			}
+
+			return true;
+		}
+
+		bool buttonRelease( const Gaffer::Path &path, const GafferUI::PathListingWidget &widget, const ButtonEvent &event )
+		{
+			auto setPath = IECore::runTimeCast<const SetPath>( &path );
+			if( !setPath )
+			{
+				return false;
+			}
+
+			const auto setName = runTimeCast<const IECore::StringData>( setPath->property( g_setNamePropertyName ) );
+			if( !setName )
+			{
+				// We only interact with locations representing sets
+				return false;
+			}
+
+			auto paths = IECore::PathMatcher();
+			const auto setMembers = setPath->getScene()->set( setName->readable() );
+			paths.addPaths( setMembers->readable() );
+			const auto selection = widget.getSelection();
+			if( std::holds_alternative<IECore::PathMatcher>( selection ) )
+			{
+				// Permit bulk editing of a selection of set names when clicking on one of the selected set names
+				const auto selectedPaths = std::get<IECore::PathMatcher>( selection );
+				if( selectedPaths.match( setPath->names() ) & IECore::PathMatcher::Result::ExactMatch )
+				{
+					for( IECore::PathMatcher::Iterator it = selectedPaths.begin(), eIt = selectedPaths.end(); it != eIt; ++it )
+					{
+						if( it->back() != setName->readable() )
```

Comment:
I don't think we can assume that the last path element is a valid set name can we? As in the code higher up, we'd need to check `property( g_setNamePropertyName )` to exclude the "folder" paths.

Reply:
Good point! I've addressed that in 1811365.